clients/mssql/dialect/dialect.go

```diff
@@ -204,12 +204,24 @@ USING %s AS %s ON %s`,
 	}
 
 	if softDelete {
-		// TODO alter this merge query to update only the __artie_delete column if OnlySetDeleteColumnMarker is true
+		// Build conditional column values for the update clause; if __artie_only_set_delete is true, we
+		// need to preserve existing values for all columns other than __artie_delete
+		colsExceptDelete, err := columns.RemoveDeleteColumnMarker(cols)
+		if err != nil {
+			return nil, err
+		}
+
+		condition := fmt.Sprintf("COALESCE(%s, 0) = 0", sql.GetQuotedOnlySetDeleteColumnMarker(constants.StagingAlias, md))
+		conditionalCols := sql.BuildConditionalColumnsUpdateFragment(colsExceptDelete, constants.StagingAlias, constants.TargetAlias, md, condition)
+		deleteCol := sql.BuildColumnsUpdateFragment([]columns.Column{columns.NewColumn(constants.DeleteColumnMarker, typing.Boolean)}, constants.StagingAlias, constants.TargetAlias, md)
+
 		return []string{baseQuery + fmt.Sprintf(`
-WHEN MATCHED %sTHEN UPDATE SET %s
+WHEN MATCHED %sTHEN UPDATE SET %s,%s
```

Comment:
Can we return 2 updates and an insert similar to Redshift?

Reply:
Btw, if you return multiple parts - Transfer will automatically detect that and wrap a tx: https://github.com/artie-labs/transfer/blob/879ae1212e440060537d1ce1c165d0bf46096233/lib/destination/dwh.go#L40

Reply:
Yeah that's an alternative. I think it will require a bigger refactor because we can't combine merge & update statements, right? So we'd have to also rewrite how we do inserts and deletes?

Out of curiosity, why do we use that approach (separate insert/update/delete statements instead of a single merge statement) for redshift? What was the limitation we ran into with redshift that a merge statement couldn't do?